### PR TITLE
ranged for over components and parameters

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: c5aa37a9a0622f3d2d1cdd3c9e1ae8cd14a4bebe).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/84951587

File: src/cpp/cpp_constructor.cpp [hunk 1]
```diff
@@ -213,18 +213,15 @@ optionalt<codet> cpp_typecheckt::cpp_constructor(
...
 
-      if(!it->get_bool(ID_from_base) &&
-         type.id()==ID_code &&
-         type.find(ID_return_type).id()==ID_constructor)
+      if(
+        !c.get_bool(ID_from_base) && type.id() == ID_code &&
+        type.find(ID_return_type).id() == ID_constructor)
       {
-        constructor_name=it->get(ID_base_name);
+        constructor_name = c.get(ID_base_name);
```
Comment (peterschrammel) [inline src/cpp/cpp_constructor.cpp:224]: `.get_base_name()` ?

File: src/cpp/cpp_destructor.cpp [hunk 2]
```diff
@@ -86,18 +86,15 @@ optionalt<codet> cpp_typecheckt::cpp_destructor(
...
 
-      if(!it->get_bool(ID_from_base) &&
-         type.id()==ID_code &&
-         type.find(ID_return_type).id()==ID_destructor)
+      if(
+        !c.get_bool(ID_from_base) && type.id() == ID_code &&
+        type.find(ID_return_type).id() == ID_destructor)
       {
-        dtor_name=it->get(ID_base_name);
+        dtor_name = c.get(ID_base_name);
```
Comment (peterschrammel) [inline src/cpp/cpp_destructor.cpp:97]: `.get_base_name()` ?

File: src/cpp/cpp_typecheck_constructor.cpp [hunk 3]
```diff
@@ -283,48 +283,47 @@ void cpp_typecheckt::default_cpctor(
...
+
+  for(const auto &mem_c : components)
   {
     // Take care of virtual tables
-    if(mem_it->get_bool("is_vtptr"))
+    if(mem_c.get_bool("is_vtptr"))
     {
       exprt name(ID_name);
-      name.set(ID_identifier, mem_it->get(ID_base_name));
+      name.set(ID_identifier, mem_c.get(ID_base_name));
```
Comment (peterschrammel) [inline src/cpp/cpp_typecheck_constructor.cpp:293]: `.get_base_name()` ?

File: src/cpp/cpp_typecheck_constructor.cpp [hunk 4]
```diff
@@ -283,48 +283,47 @@ void cpp_typecheckt::default_cpctor(
...
-      || mem_it->get_bool(ID_is_static)
-      || mem_it->type().id()==ID_code)
-        continue;
+    if(
+      mem_c.get_bool("from_base") || mem_c.get_bool(ID_is_type) ||
+      mem_c.get_bool(ID_is_static) || mem_c.type().id() == ID_code)
+      continue;
 
-    irep_idt mem_name=mem_it->get(ID_base_name);
+    irep_idt mem_name = mem_c.get(ID_base_name);
```
Comment (peterschrammel) [inline src/cpp/cpp_typecheck_constructor.cpp:326]: `.get_base_name()` ?

File: src/cpp/cpp_typecheck_constructor.cpp [hunk 5]
```diff
@@ -563,27 +562,24 @@ void cpp_typecheckt::check_member_initializers(
...
     bool ok=false;
 
-    for(struct_typet::componentst::const_iterator
-        c_it=components.begin();
-        c_it!=components.end();
-        c_it++)
+    for(const auto &c : components)
     {
-      if(c_it->get(ID_base_name)!=base_name)
+      if(c.get(ID_base_name) != base_name)
```
Comment (peterschrammel) [inline src/cpp/cpp_typecheck_constructor.cpp:567]: `.get_base_name()` ?

File: src/cpp/cpp_typecheck_constructor.cpp [hunk 6]
```diff
@@ -743,12 +738,11 @@ void cpp_typecheckt::full_member_initialization(
...
 
-          for(struct_typet::componentst::const_iterator c_it =
-              components.begin(); c_it!=components.end(); c_it++)
+          for(const auto &c : components)
           {
-            if(c_it->get(ID_base_name)==base_name &&
-               c_it->get(ID_type)!=ID_code &&
-               !c_it->get_bool(ID_is_type))
+            if(
+              c.get(ID_base_name) == base_name && c.get(ID_type) != ID_code &&
```
Comment (peterschrammel) [inline src/cpp/cpp_typecheck_constructor.cpp:744]: `.get_base_name()` ?

File: src/cpp/cpp_typecheck_constructor.cpp [hunk 7]
```diff
@@ -283,48 +283,47 @@ void cpp_typecheckt::default_cpctor(
...
       exprt var=virtual_table_symbol_var.symbol_expr();
       address_of_exprt address(var);
-      assert(address.type()==mem_it->type());
+      assert(address.type() == mem_c.type());
 
       already_typechecked(address);
 
       exprt ptrmember(ID_ptrmember);
-      ptrmember.set(ID_component_name, mem_it->get(ID_name));
+      ptrmember.set(ID_component_name, mem_c.get(ID_name));
```
Comment (peterschrammel) [inline src/cpp/cpp_typecheck_constructor.cpp:313]: `.get_name()` ?

File: src/cpp/cpp_typecheck_constructor.cpp [hunk 8]
```diff
@@ -283,48 +283,47 @@ void cpp_typecheckt::default_cpctor(
...
     }
 
-    if( mem_it->get_bool("from_base")
-      || mem_it->get_bool(ID_is_type)
-      || mem_it->get_bool(ID_is_static)
-      || mem_it->type().id()==ID_code)
-        continue;
+    if(
+      mem_c.get_bool("from_base") || mem_c.get_bool(ID_is_type) ||
+      mem_c.get_bool(ID_is_static) || mem_c.type().id() == ID_code)
```
Comment (smowton) [inline src/cpp/cpp_typecheck_constructor.cpp:323]: Suggest add braces since the condition spans multiple lines

**Review by smowton — APPROVED** [review]

**Review by owen-mc-diffblue — APPROVED** [review]

**Comment by kroening** [comment]
@peterschrammel I would do the .get_base_name() refactoring in a separate PR; people have a very specific idea of the remit of a 'ranged-for' refactoring.

**Review by tautschnig — APPROVED** [review]

**Comment by tautschnig** [comment]
Just for the record: I'm working on the further cleanup suggested by @peterschrammel.

**Pull request merged**